vmdb/app/models/service_orchestration.rb

```diff
@@ -0,0 +1,56 @@
+class ServiceOrchestration < Service
+  include ServiceOrchestrationMixin
+
+  attr_writer :stack_options  # this will override all existing stack options
+  attr_writer :stack_name
+
+  def stack_options
+    @stack_options ||= build_stack_options
+  end
+
+  def stack_name
+    @stack_name ||= OptionConverter.get_stack_name(options[:dialog] || {})
+  end
+
+  def orchestration_stack_status
+    orchestration_manager.stack_status(stack_name)
+  rescue MiqException::MiqOrchestrationStatusError => err
+    # naming convention requires status to end with "failed"
+    return "check_status_failed", err.message
+  end
+
+  def deploy_orchestration_stack
+    save_stack_options
+    orchestration_manager.stack_create(stack_name, orchestration_template, stack_options)
+    nil  # if no exception
+  rescue MiqException::MiqOrchestrationProvisionError => err
+    err.message
+  end
+
+  private
+
+  def build_stack_options
+    # manager from dialog_options overrides the one copied from service_template
+    manager_from_dialog = OptionConverter.get_manager(options[:dialog] || {})
+    self.orchestration_manager = manager_from_dialog if manager_from_dialog
```

Comment:
You do not need to include `self` here.  There are a few other unneeded `self` references in this commit that can be removed.